static/src/javascripts/projects/atoms/services/viewport.js

```diff
@@ -0,0 +1,72 @@
+// @flow
```

Comment:
can you explain what this for? does it achieve anything `IntersectionObserver` does not?

Reply:
Nope. It is only to have a consistent API between dotcom and apps, where IO is not polyfilled and I didn't want to go all the way to do that, at least not at first.